SIPS/sip-59.md

```diff
@@ -0,0 +1,141 @@
+---
+sip: 59
+title: Delegated Migrator
+status: Proposed
+author: Justin J Moses (@justinjmoses)
+discussions-to: <https://discordapp.com/invite/AEdUHzt>
+
+created: 2020-05-11
+---
+
+<!--You can leave these HTML comments in your merged SIP and delete the visible duplicate text guides, they will not appear and may be helpful to refer to if you edit it again. This is the suggested template for new SIPs. Note that an SIP number will be assigned by an editor. When opening a pull request to submit your SIP, please use an abbreviated title in the filename, `sip-draft_title_abbrev.md`. The title should be 44 characters or less.-->
+
+## Simple Summary
+
+<!--"If you can't explain it simply, you don't understand it well enough." Provide a simplified and layman-accessible explanation of the SIP.-->
+
+Make upgrades of Synthetix as seamless as possible
```

Comment:
I'm leaning towards new modifier ownerOrpDAO for SCCPs / configurable properties to allow the pDAO to configure the way it does now.
- Set Synth exchange pricing (For Differential Fees)
- Add/Edit Delete Rewards Distributions (quite a bit lately with sETH, iETH & ArbRewarder)
- Purging and Reseting iSynths
- Manual Rewards trials for Staking Rewards (was curve, currently iETH)
- C-ratio
- Claim Fees Threshold
- Escrowing SNX
- EtherCollateral Config (pause, setIssueLimit, setInterestRate, setMinLoanSize etc)
- Depot Config (pause, setMaxEthPurchase, setMinimumDepositAmount etc)

Reply:
I agree, 

It would be nice if the DelegateMigrator has the ability to execute SCCP's after a vote / consensus is reached in the future.

In the short term, an admin role assigned to the pDAO (that will replace what the owner is currently doing) for executing the above SCCP / adjustments that needs to be made to the system. 

This will allow the DelegateMigrator to focus on voting on contract upgrades and executing the contract upgrades to wire up system.

Reply:
So `purge` should be added to a separate access control. For the other functions, longer term I suggest using a single `EternalStorage` for all these params and then we can limit interaction that way. Though that being said, @jacko125 has suggested a way to whitelist certain function signatures and targets we could consider...

Reply:
The problem with the new modifier @hav-noms suggested - `ownerOrpDAO` - is that there are some immutable contracts to consider - like `SynthetixState.setIssuanceRatio()`... so it isn't really a viable solution.

Reply:
SIP-64 addresses this with `SystemSettings`. See https://github.com/Synthetixio/synthetix/pull/616